### ISSUE TITLE
better error message for e.g. `convert(Tuple{NTuple{2, Int}, Int}, (3,3,3))`

### ISSUE THREAD

**Issue by goretkin** [JuliaLang/julia#36573]
The `convert` methods are defined using the structural recursion on tuples, so the error message comes from a nested call:

```julia
julia> convert(Tuple{NTuple{2, Int}, Int}, (3,3,3))
ERROR: MethodError: Cannot `convert` an object of type Int64 to an object of type Tuple{Int64,Int64}
```

It might be better to produce the following error message, if possible.
````ERROR: MethodError: Cannot `convert` an object of type Tuple{Int64,Int64,Int64} to an object of type Tuple{Tuple{Int64,Int64},Int64}````

**Comment by vtjnash** [comment]
I'd attempted this in https://github.com/JuliaLang/julia/pull/32427, but had gotten hung up on a type-system bug that made it segfault when I tried to detect this case. I think I've found a workaround.

**Comment by ViralBShah** [comment]
With julia master:

```
julia> convert(Tuple{NTuple{2, Int}, Int}, (3,3,3))
ERROR: MethodError: Cannot `convert` an object of type 
  Tuple{Int64,Int64{},Int64} to an object of type 
  Tuple{Tuple{Int64, Int64},Int64{}}

```

**Comment by vtjnash** [comment]
Fixed by #32427
